benchmark/Streamly/Benchmark/Data/Unfold.hs

```diff
@@ -131,57 +410,122 @@ zipWithM size start =
-        -- have microseconds as the minimum unit before uncommenting this.
-        -- , benchIO "dropOne" $ dropOne size
-        , benchIO "dropAll" $ dropAll size
-        , benchIO "dropWhileMTrue" $ dropWhileMTrue size
-        , benchIO "dropWhileMFalse" $ dropWhileMFalse size
-        ]
+o_1_space_generation_resource :: Int -> [Benchmark]
+o_1_space_generation_resource size =
+    [ bgroup
+          "generation/resource"
+          [ benchIO "fromStream" $ fromStream size
+          , benchIO "fromStreamK" $ fromStreamK size
+          -- XXX INVESTIGATE, see the definition
+          -- , benchIO "fromStreamD" $ fromStreamD size
+          -- Very small benchmarks, reporting in ns
+          -- , benchIO "nilM" $ nilM size
+          , benchIO "consM" $ consM size
+          -- , benchIO "effect" $ effect size
+          -- , benchIO "singletonM" $ singletonM size
+          -- , benchIO "singleton" $ singleton size
+          -- , benchIO "identity" $ identity size
+          -- , benchIO "const" $ const size
+          , benchIO "unfoldrM" $ unfoldrM size
+          , benchIO "fromList" $ fromList size
+          , benchIO "fromListM" $ fromListM size
+          -- Unimplemented
+          -- , benchIO "fromSVar" $ fromSVar size
+          -- , benchIO "fromProducer" $ fromProducer size
+          ]
     ]
 
-o_1_space_serial_combination :: Int -> [Benchmark]
-o_1_space_serial_combination size =
-    [ bgroup "combination"
-        [ benchIO "zipWithM" $ zipWithM size
-        ]
+
+o_1_space_generation_specialized :: Int -> [Benchmark]
+o_1_space_generation_specialized size =
+    [ bgroup
+          "generation/specialized"
+          [ benchIO "replicateM" $ replicateM size
+          , benchIO "repeatM" $ repeatM size
+          , benchIO "iterateM" $ iterateM size
+          , benchIO "fromIndicesM" $ fromIndicesM size
+          , benchIO "enumerateFromStepIntegral" $ enumerateFromStepIntegral size
+          , benchIO "enumerateFromToIntegral" $ enumerateFromToIntegral size
+          , benchIO "enumerateFromIntegral" $ enumerateFromIntegral size
+          , benchIO "enumerateFromStepNum" $ enumerateFromStepNum size
+          , benchIO "numFrom" $ numFrom size
+          , benchIO "enumerateFromToFractional" $ enumerateFromToFractional size
+          ]
     ]
 
-o_1_space_serial_outerProduct :: Int -> [Benchmark]
-o_1_space_serial_outerProduct size =
-    [ bgroup "outer-product"
-        [ benchIO "toNull" $ toNull size
-        , benchIO "toNull3" $ toNull3 size
-        , benchIO "concat" $ concat size
-        , benchIO "breakAfterSome" $ breakAfterSome size
-        ]
+o_1_space_transformation :: Int -> [Benchmark]
+o_1_space_transformation size =
+    [ bgroup
+          "transformation"
+          [ benchIO "map" $ map size
+          , benchIO "mapM" $ mapM size
+          , benchIO "mapMWithInput" $ mapMWithInput size
+          , benchIO "takeWhileM" $ takeWhileM size
```

Comment:
We can make a `o_1_space_filtering` group for all filtering ops, like we have in serial stream benchmarks.